kong/core/router.lua

```diff
@@ -637,12 +637,25 @@ function _M.new(apis)
     end
 
 
-    if not uri_root and api_t.strip_uri_regex then
-      local err
-      new_uri, err = re_sub(uri, api_t.strip_uri_regex, "/$1", "ajo")
-      if not new_uri then
-        log(ERR, "could not strip URI: ", err)
-        return
+    if uri_root then
+      new_uri = uri
+
+    else
+      local s = find(uri, "?", 2, true)
+      if s then
+        uri = sub(uri, 1, s - 1)
```

Comment:
We need to strip the `uri` prior to `find_api` call. The reason is that as this stands now, we would greatly increase the risk of cache miss because the same URL path may have different querystring parameters, hence resulting in a different cache key.